integration_tests/containers/test_cleanbuild.py

```diff
@@ -0,0 +1,44 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015-2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import subprocess
+
+from testtools.matchers import FileExists
+
+import integration_tests
+
+
+class CleanbuildTestCase(integration_tests.TestCase):
+
+    def run_snapcraft_cleanbuild(self, project_dir):
+        if project_dir:
+            self.copy_project_to_cwd(project_dir)
+
+        command = ['sudo', self.snapcraft_command, '-d', 'cleanbuild']
```

Comment:
I like the `containers` category... although I'm wondering how this will look in the context of consolidating integration and unit tests. In unit tests, we have `commands` here, since how you run snapcraft is the relevant bit, not what they do in the background. What's your thoughts on this?

These questions are popping up in my head:
* you're not using `run_snapcraft` anymore
* `sudo` is needed?
* you're using `subprocess.Popen` instead of `subprocess.check_output`

First of all, I'd consider it a good thing to make the use of subprocess more obvious in the test case, since I recently confused an integration test with a unit test even after having written several ones... but I also feel logic like this should be re-used. Maybe something like `snapcraft_process` that returns the process?

And comments on why you're manually handling stdout and why sudo is used would be appreciated.